onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -904,6 +905,194 @@
                                   updateOutputElemType(ctx, 0, ONNX_NAMESPACE::TensorProto::BOOL);
                                 }));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // All inputs shapes are required
+  if (!hasNInputShapes(ctx, 3)) {
+    return;
+  }
+
+  // Assuming image_shape has correct spatial dimensions and reused for next validation steps
+  //       An alternative is get the the number of spatial dimensions as an input
+  if (ctx.getInputType(1)->tensor_type().shape().dim_size() != 1) {
```

Comment:
>ctx.getInputType(1)->tensor_type( [](http://example.com/codeflow?start=6&length=33)

nit: suggest eliminating long sequences of repeated code.

Reply:
What would be your suggestion? Creating temporary variables to hold intermediate states?
In ONNX repo, the sequence above is acceptable if are not longer than 80 chars.

Reply:
a single auto* tensor_type would do a great deal to aid in readability.